core/connection/manager.go

```diff
@@ -206,29 +223,48 @@ func (manager *connectionManager) startConnection(consumerID identity.Identity,
 		SessionInfo: manager.sessionInfo,
 	})
 
-	cancel = append(cancel, func() {
+	manager.cleanup = append(manager.cleanup, func() {
 		manager.eventPublisher.Publish(SessionEventTopic, SessionEvent{
 			Status:      SessionEndedStatus,
 			SessionInfo: manager.sessionInfo,
 		})
 	})
 
+	return s, paymentInfo, nil
+}
+
+func (manager *connectionManager) startConnection(
+	connection Connection,
+	consumerID identity.Identity,
+	proposal market.ServiceProposal,
+	params ConnectParams,
+	sessionDTO session.SessionDto,
```

Comment:
I'd name it just `session` without `DTO`.

Reply:
conflicts with the package name

Reply:
can be `sessionData`